store/src/java/com/zimbra/cs/mailbox/ContactAutoComplete.java

```diff
@@ -756,6 +747,48 @@ public void addMatchedContacts(String query, Map<String, ? extends Object> attrs
         }
     }
 
+    /**
+     * Identifies the eligible results for GAL Autocomplete for an account
+     * @param attrs
+     * @return
+     * @throws ServiceException
+     */
+    protected List<String> extractEligibleEmailsListForAccount(Map<String, ?> attrs) throws ServiceException {
+        List<String> eligibleEmailsForAccount = new ArrayList<>();
+        String primaryEmail = getFieldAsString(attrs, ContactConstants.A_email);
+        Account account = Provisioning.getInstance().get(Key.AccountBy.name, primaryEmail);
+        populateEmailsList(attrs, eligibleEmailsForAccount, account);
+        return eligibleEmailsForAccount;
+    }
+
+    /**
+     * populats the eligible
+     * @param attrs
+     * @param eligibleEmailsForAccount
+     * @param account
+     * @throws ServiceException
+     */
+    private void populateEmailsList(Map<String, ?> attrs, List<String> eligibleEmailsForAccount, Account account)
+            throws ServiceException {
+        for (String emailKey : mEmailKeys) {
+            if (null != account) {
+                if (ContactConstants.A_email.equals(emailKey) && account.isHideInGal()) {
+                    continue;
+                }
+                //when account is null, its external contact.Below alias logic is not applicable.
```

Comment:
add space after // here and other places for comments

Reply:
also this comment should be for line #774

Reply:
fixed